workers/gossip-server/src/heartbeat.rs

```diff
@@ -367,48 +272,25 @@ impl HeartbeatTimer {
     async fn inter_cluster_execution_loop(
         job_queue: TokioSender<GossipServerJob>,
         wait_period: Duration,
-        global_state: RelayerState,
-    ) -> GossipError {
-        let mut peer_index = 0;
-        let local_cluster = global_state.local_cluster_id.clone();
+        global_state: State,
+    ) -> Result<(), GossipError> {
+        let cluster_id = global_state.get_cluster_id()?;
 
         loop {
-            let (peer_count, next_peer_id) = {
-                // Enqueue a heartbeat job for each known peer
-                let peer_info_locked = global_state.read_peer_index().await;
-                let next_peer = peer_info_locked.nth(peer_index).await;
-
-                // Skip if we have overflowed the list or if the next peer is in the local
-                // peer's cluster; a separate timer will enqueue intra-cluster
-                // heartbeats at a faster rate
-                let mut next_peer_id = None;
-                if let Some(peer_info) = next_peer {
-                    if peer_info.get_cluster_id() != local_cluster {
-                        next_peer_id = Some(peer_info.get_peer_id())
-                    }
-                }
+            // Get all peers not in the local peer's cluster
+            let mut peers = global_state.get_peer_info_map()?;
+            peers.retain(|_, peer_info| peer_info.get_cluster_id() != cluster_id);
```

Comment:
is there a helper for this to the tune of `global_state.get_external_peers()`? if not it could be pretty neat here, but w/e